src/components/Filter.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { useContext } from 'react';
+import cn from 'classnames';
+import { TodoContext } from './TodoContext';
+import { FilterAction } from '../types/Actions';
+import { deleteTodo } from '../api/todos';
+
+export const Filter: React.FC = () => {
+  const { todos, filter, setFilter } = useContext(TodoContext);
+  const handleFilterChange = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    type: FilterAction,
+  ) => {
+    event.preventDefault();
+    setFilter(type);
+  };
+
+  const hasCompleteTodo = todos.filter(todo => todo.completed === true);
+
+  const showTodos = todos.filter(todo => !todo.completed).length;
+
+  const handleDeletCompleteTodos = () => {
+    for (const item of hasCompleteTodo) {
+      deleteTodo(item.id);
+    }
+  };
+
+  return (
+    <>
+      {todos.length > 0 && (
```

Comment:
```suggestion
      {!!todos.length && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**